tests/pytests/unit/states/test_service.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import salt.states.service as service
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {service: {}}
+
+
+def test_when_service_available_in_dunder_salt_then_it_should_be_used():
+    expected_ret = {}
+    with patch.dict(service.__salt__, {"service.available": lambda x: True}):
+        ret = {}
+        assert service._available(name="fnord", ret=ret)
+        assert ret == expected_ret
+
+
+def test_when_service_get_all_in_dunder_salt_it_should_be_used_as_fallback():
+    avail_services = ["fnord", "foo", "bar"]
+    expected_ret = {}
+    with patch.dict(service.__salt__, {"service.get_all": lambda: avail_services}):
+        ret = {}
+        assert service._available(name=avail_services[0], ret=ret)
+        assert ret == expected_ret
+
+
+def test_when_neither_available_or_get_all_in_dunder_salt_then_result_should_be_False():
+    expected_ret = {
+        "result": False,
+        "comment": "The named service fnord is not available",
+    }
+    ret = {}
+    assert not service._available(name="fnord", ret=ret)
+    assert ret == expected_ret
+
+
+@pytest.mark.parametrize(
+    "test_func", [service._enable, service._disable],
+)
+def test_when_skip_verify_enable_should_ignore_if_service_is_available(test_func):
+    with patch(
+        "salt.states.service._available", side_effect=Exception("Should not be raised")
+    ):
+        test_func("fnord", "fnord", skip_verify=True)
+
+
+@pytest.mark.parametrize(
+    "test_func", [service._enable, service._disable],
+)
+def test_when_not_skip_verify_and_available_raises_CommandExecutionError_result_should_be_false_with_exec_strerror(
```

Comment:
Could we instead have shorter function names and if needed a docstring explaining the detail, instead of trying to squeeze all that into the test function name?